apps/seqan_flexbar/general_processing.h

```diff
@@ -483,474 +483,240 @@ void processN(TSeqs& seqs, TIds& ids, TSeqs& seqsRev, TIds& idsRev, TMulti& mult
-    StringSet<bool> rem;
-
-    resize(rem, limit);
-    if (head > 0 && tail > 0)
-    {
-        SEQAN_OMP_PRAGMA(parallel for default(shared) private(i) schedule(static))
-        for (i = 0; i < limit; ++i)
-        {
-            if (length(seqs[i]) > (head + tail))
-            {
-                erase(seqs[i], 0, head);
-                erase(seqs[i], length(seqs[i]) - tail, length(seqs[i]));
-                if (length(seqs[i]) >= min)
-                {
-                    rem[i] = false;
-                }
-                else
-                {
-                    rem[i] = true;
-                }
-            }
-            else
-            {
-                rem[i] = true;
-            }
-        }
-    }
-    else if (head > 0)
-    {
-        SEQAN_OMP_PRAGMA(parallel for default(shared) private(i) schedule(static))
-        for (i = 0; i < limit; ++i)
-        {
-           if (length(seqs[i]) > head)
-            {
-                erase(seqs[i], 0, head);
-                if (length(seqs[i]) >= min)
-                {
-                    rem[i] = false;
-                }
-                else
-                {
-                    rem[i] = true;
-                }
-            }
-            else
-            {
-                rem[i] = true;
-            }
-        }
-    }
-    else if (tail > 0)
-    {
-        SEQAN_OMP_PRAGMA(parallel for default(shared) private(i) schedule(static))
-        for (i = 0; i < limit; ++i)
-        {
-            if (length(seqs[i]) > tail)
-            {
-                erase(seqs[i], length(seqs[i]) - tail, length(seqs[i]));
-                if (length(seqs[i]) >= min)
-                {
-                    rem[i] = false;
-                }
-                else
-                {
-                    rem[i] = true;
-                }
-            }
-            else
-            {
-                rem[i] = true;
-            }
-        }
-    }
-    else
-    {
-        SEQAN_OMP_PRAGMA(parallel for default(shared) private(i) schedule(static))
-        for (i = 0; i < limit; ++i)
-        {
-            if (length(seqs[i]) >= min)
-            {
-                rem[i] = false;
-            }
-            else
-            {
-                rem[i] = true;
-            }
-        }
-    }
-    unsigned  ex = 0;
-    for (int j = length(rem) - 1; j >= 0; --j)
-    {
-        if (rem[j])
-        {
-            swap(seqs[j], seqs[limit - ex - 1]);
-            swap(ids[j], ids[limit - ex - 1]);
-            ++ex;
-        }
-    }
-    if (ex != 0)
-    {
-        resize(seqs, limit - ex);
-        resize(ids, limit - ex);
-        stats.removedSeqsShort += ex;
-    }
+	assert(!seqsVector.empty() && !idsVector.empty());
+	assert(seqsVector.size() == idsVector.size());
+	unsigned ex = 0;
+	unsigned limit = length(**seqsVector.begin());
+	bool demultiplex = (length(demultiplexParams.multiplexFile) > 0);
+	for (int j = limit - 1; j >= 0; j--)
+	{
+		if (rem[j])
```

Comment:
It does what this code did before:
-    unsigned ex = 0;
-    for (int j = length(rem) - 1; j >= 0; --j)
-    {
-        if (rem[j])
-        {
-            swap(seqs[j], seqs[limit - ex - 1]);
-            swap(seqsRev[j], seqsRev[limit - ex - 1]);
-            swap(ids[j], ids[limit - ex - 1]);
-            swap(idsRev[j], idsRev[limit - ex - 1]);
-            ++ex;
-        }
-    }
-    if (ex != 0)
-    {
-        resize(seqs, limit - ex);
-        resize(seqsRev, limit - ex);
-        resize(ids, limit - ex);
-        resize(idsRev, limit - ex);
-        stats.removedSeqsShort += (2 * ex);
-    }